src/app/loadout-builder/process-worker/process.ts

```diff
@@ -280,110 +126,107 @@ export function process(
+            const stats = [
+              modStatsInStatOrder[0] +
+                helmStats[0] +
+                gauntStats[0] +
+                chestStats[0] +
+                legStats[0] +
+                classItemStats[0],
+              modStatsInStatOrder[1] +
+                helmStats[1] +
+                gauntStats[1] +
+                chestStats[1] +
+                legStats[1] +
+                classItemStats[1],
+              modStatsInStatOrder[2] +
+                helmStats[2] +
+                gauntStats[2] +
+                chestStats[2] +
+                legStats[2] +
+                classItemStats[2],
+              modStatsInStatOrder[3] +
+                helmStats[3] +
+                gauntStats[3] +
+                chestStats[3] +
+                legStats[3] +
+                classItemStats[3],
+              modStatsInStatOrder[4] +
+                helmStats[4] +
+                gauntStats[4] +
+                chestStats[4] +
+                legStats[4] +
+                classItemStats[4],
+              modStatsInStatOrder[5] +
+                helmStats[5] +
+                gauntStats[5] +
+                chestStats[5] +
+                legStats[5] +
+                classItemStats[5],
+            ];
+
+            // TODO: avoid min/max?
+            const tiers = [
+              Math.min(Math.max(Math.floor(stats[0] / 10), 0), 10),
+              Math.min(Math.max(Math.floor(stats[1] / 10), 0), 10),
+              Math.min(Math.max(Math.floor(stats[2] / 10), 0), 10),
+              Math.min(Math.max(Math.floor(stats[3] / 10), 0), 10),
+              Math.min(Math.max(Math.floor(stats[4] / 10), 0), 10),
+              Math.min(Math.max(Math.floor(stats[5] / 10), 0), 10),
+            ];
+            const totalTier = tiers[0] + tiers[1] + tiers[2] + tiers[3] + tiers[4] + tiers[5];
+
+            // Check whether the set exceeds our stat constraints
             let statRangeExceeded = false;
-            for (const statHash of statOrder) {
-              const value = stats[statHash];
-              const tier = statTier(value);
-
-              // Update our global min/max for this stat
-              const range = statRanges[statHash];
-              if (value > range.max) {
-                range.max = value;
-              }
-              if (value < range.min) {
-                range.min = value;
-              }
-
-              const filter = statFilters[statHash];
-              if (!filter.ignored) {
-                if (tier > filter.max || tier < filter.min) {
-                  statRangeExceeded = true;
-                }
-                totalTier += tier;
+            for (let index = 0; index < 6; index++) {
+              const tier = tiers[index];
+              const filter = statFiltersInStatOrder[index];
+              if (!filter.ignored && (tier > filter.max || tier < filter.min)) {
+                statRangeExceeded = true;
               }
             }
```

Comment:
Would unrolling this loop make any difference?

Reply:
It could, but it's relatively difficult to do so.